viewing_party/party.py

```diff
@@ -1,23 +1,156 @@
+from statistics import mean
+
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    movie_dict = {}
+    while title and genre and rating:
+        movie_dict["title"] = title
+        movie_dict["genre"] = genre
+        movie_dict["rating"] = rating
+        return movie_dict
+
+    return None
+
+def add_to_watched(user_data, movie):
+    # user_data ={
+    #     "watched" : [{movie}]
+    # }
+
+    # movie = {
+    #     "title" :
+    #     "genre" :
+    #     "rating":
+    # }
+    # watched_list = user_data["watched"]
+    # while movie not in watched_list:
+    #     watched_list.append(movie)
+    user_data["watched"].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, title):
+    # watchlist = add_to_watchlist(user_data, movie)
+    watchlist = user_data["watchlist"]
+    # watched_list = add_to_watchlist(user_data, movie)
+    watched_list = user_data["watched"]
+
+    for movie in watchlist:
+        if movie["title"] == title:
+            movie_list = watchlist.pop()
+            watched_list.append(movie_list)
+    return user_data
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+from statistics import mean
 
+def get_watched_avg_rating(user_data):
+    watched_list = user_data["watched"]
+    # watchlist = user_data["watchlist"]
+    rating_list = []
+    num_movie = len(watched_list)
+    if num_movie == 0:
+        return 0
+    for movie in watched_list:
+        # for rating in movie["rating"]:
+            rating_list.append(movie["rating"])
+    # return rating_list
+    return mean(rating_list)
+
+def get_most_watched_genre(user_data):
+    watched_list = user_data["watched"]
+    genre_list = []
+    
+    while len(watched_list) > 0:
+        for movie in watched_list:
+            genre_list.append(movie["genre"])

```

Comment:
Fun fact- this genre_list can also be initialized with a list comprehension:
`genre_list = [movie["genre"] for movie in watched_list]`